runtime/gc_base/StringTable.cpp

```diff
@@ -952,21 +937,11 @@ j9gc_internString(J9VMThread *vmThread, j9object_t sourceString)
 	candidate = *candidatePtr;
 
 	if ((NULL != candidate) && stringHashEqualFn(&candidate, &sourceString, vm)) {
-		if (isMetronome) {
-#if defined(J9VM_GC_REALTIME) 
-			/*
-			 * This can only be used if the current candidate pointer is live.
-			 * Pass in candidate twice since we only have one string.
-			 */
-			if (FALSE != j9gc_objaccess_checkStringConstantsLive(vm, candidate, candidate)) {
-				Trc_MM_stringTableCacheHit(vmThread, candidate);
-				return candidate;
-			}
-#else
-			Trc_MM_stringTableCacheHit(vmThread, candidate);
-			return candidate;
-#endif
-		} else {
+		/*
+		 * This can only be used if the current candidate pointer is live.
+		 * Pass in candidate twice since we only have one string.
+		 */
+		if (FALSE != checkStringConstantsLive(vm, candidate, candidate)) {
```

Comment:
You can probably just do !checkStringConstantsLive, It's the style used elsewhere in this file too (or change the other spot to use this style, comparing with a BOOL const)

Reply:
Changed to ` !checkStringConstantsLive`

Reply:
That's not the same condition (`!checkStringConstantsLive()` is equivalent to `FALSE == checkStringConstantsLive()`). Is the intent to invert the test?

Reply:
No, it was not intention to change the behavior. Thanks for catching this @keithc-ca.

Reply:
I'll change it to `if (checkStringConstantsLive())`